### DISCUSSION TITLE
How to separate `devDependencies`?

### DISCUSSION THREAD

**Discussion by Shinigami92** [renovatebot/renovate#11310] [Request Help] [answered]
I want create a config that runs only on fridays between 9am to 7pm
Also I want to have a separate PR that bumps all my `devDependencies` to the newest versions but using `~` (tilde)

For the one prod `dependency` I would like to use `^` but I don't want to increase the version and stay at `^5.0.0`. The `yarn.lock` is okay to have it on `^5.0.1`.

Due to a problem in `prettier-plugin-organize-imports` I currently need to skip and stay on `2.2.0`

My current config:

```jsonc
{
	"extends": ["config:base", "group:allNonMajor"],
	"schedule": ["after 9am and before 7pm on friday"],
	"timezone": "Europe/Berlin",
	"labels": ["dependencies"],
	"reviewers": ["Shinigami92"],
	"lockFileMaintenance": { "enabled": true },
	"pin": false,
	"rangeStrategy": "replace",
	"ignoreDeps": ["prettier-plugin-organize-imports"],
	"packageRules": [
		{
			"matchDepTypes": ["devDependencies"],
			"groupName": "all devDependencies"
		}
	]
}
```

https://github.com/prettier/plugin-pug/pull/280/files#diff-f82f7d36a61d22f640c25bdb8b0435bf9cb38679d99d5f460753fa668ee7cdb3

---

What I'm doing wrong? Why doesn't it work and show me that it will create a separate PR for `devDependencies`?

**Comment by rarkins** [comment, marked as answer]
First of all, I recommend you remove `"group:allNonMajor"` from your extends and instead create a rule for it so that it's clearer about which dependency goes where.

Renovate by default won't upgrade any dependency if the new version already satisfies the existing range. If you want to have ranges in your package.json but bump them so that it's always the latest version then you need `rangeStrategy=bump`.

**Reply by Shinigami92** [reply]
I updated it to:

```jsonc
{
	"extends": ["config:base"],
	"schedule": ["after 9am and before 7pm on friday"],
	"timezone": "Europe/Berlin",
	"labels": ["dependencies"],
	"reviewers": ["Shinigami92"],
	"lockFileMaintenance": { "enabled": true },
	"pin": false,
	"rangeStrategy": "replace",
	"ignoreDeps": ["prettier-plugin-organize-imports"],
	"packageRules": [
		{
			"matchPackagePatterns": ["*"],
			"matchUpdateTypes": ["minor", "patch"],
			"groupName": "all non-major dependencies",
			"groupSlug": "all-minor-patch"
		},
		{
			"matchDepTypes": ["devDependencies"],
			"groupName": "all devDependencies",
			"groupSlug": "all-dev-dependencies",
			"rangeStrategy": "bump"
		}
	]
}
```

Seems to work :thinking: for now...
